mediation_fees.rst

```diff
@@ -107,115 +107,124 @@ easier to let them define a *per-hop* proportional mediation fee (called
 .. math::
 
-    {fee}_{in} = a - c = qa + f + i(-a)
-
-From (2) follows:
+    i(x) := \mathit{IP}(t + x) - \mathit{IP}(t)
 
-.. math::
-
-    \begin{split}
-    c &= b + bq + f + i(b) \\
-    b &= \frac{c - f - i(b)}{1+q}
-    \end{split}
+and :math:`t` is the balance of the that channel. The amount :math:`x`
+is positive for incoming channels and negative for outgoing channels to
+reflect the change in balance. The values :math:`\mathit{flat}`,
+:math:`\mathit{q}` and the function :math:`\mathit{IP}` comprise the
+fee schedule for the channel.
 
-This leads to
+To get the mediation fee for a single mediator, we have to sum up the fees
+across both involved channels:
 
 .. math::
 
-    {fee}_{out} = c - b = c - \frac{c - f - i(b)}{1+q}
-
-Here one can see that the calculation depends on both :math:`b` and :math:`c`. This formula doesn't have a symbolic solution for arbitrary functions :math:`i(x)`.
+   \begin{align}
+   \mathit{fee}_m & := round(\mathit{fee}_{\mathit{in}}(\mathit{x_{in}}) + \mathit{fee}_{\mathit{out}}(-\mathit{x_{out}})) & (1)\\
+   \mathit{fee}_m & := x_{in} - x_{out} & (2)
+   \end{align}
 
-We approximate the solution by calculating :math:`b \approx b' = \frac{c - f}{1+q}` and than use that to solve for :math:`b` (which is the first iteration towards the solution which assumes :math:`i = 0`):
+When the mediator has enabled fee capping (which is the default), the result will not go below zero.
 
 .. math::
 
-    {fee}_{out} = c - b \approx c - \frac{c - f - i(b')}{1+q}
+   \mathit{fee}_{m\_capped} := max(\mathit{fee}_m, 0)
 
-Backward calculation (as in the PFS)
-------------------------------------
 
-In the case of fee calculation in the PFS, only :math:`b` is known and it needs to calculate :math:`c` and :math:`a`.
 
-From (2) follows:
+Forward calculation (used in the mediator)
+------------------------------------------
 
-.. math::
-
-    {fee}_{out} = c - b = bq + f + i(b)
-
-From (1) follows:
+A mediator already knows :math:`x_{in}`, but knows neither :math:`x_{out}` nor
+:math:`\mathit{fee}_m` both of which can't be calculated directly without
+knowing the other.  So instead of a directly calculating them, we solve the
+equation we get by equating (1) and (2).
 
 .. math::
 
-    {fee}_{in} = a - c = \frac{c + f + i(-a)}{1-q} - c
-
-Here the same approximation approach is used for the imbalance fee. The approximation :math:`i(-a')` with :math:`a' = \frac{c + f}{1+q}` is used in the symbolic solution.
+   x_{in} - x_{out} = round(\mathit{fee}_{\mathit{in}}(\mathit{x_{in}}) + \mathit{fee}_{\mathit{out}}(-\mathit{x_{out}}))
 
+The only unknown in this equation is :math:`x_{out}` which makes this
+equivalent to finding the zero of
+   
 .. math::
+   f(x_{out}) = round(\mathit{fee}_{\mathit{in}}(\mathit{x_{in}}) + \mathit{fee}_{\mathit{out}}(-\mathit{x_{out}})) - x_{in} + x_{out}
 
-    {fee}_{in} = a - c \approx \frac{c + f + i(-a')}{1-q} - c
+Due to the constraints on the fee schedule, this function is monotonically
```

Comment:
Do we define these constraints somewhere?

Edit: There's something below, can we link to that? (https://external-builds.readthedocs.io/html/raiden-network-specification/322/mediation_fees.html#requirements )

Reply:
We haven't collected the right ones for this case. I can add a footnote. I think it's just `max_prop_fee + 2 * max_slope << 1`